src/main/java/gim/model/exercise/Weight.java

```diff
@@ -50,4 +50,8 @@ public int hashCode() {
         return value.hashCode();
     }
 
+    @Override
+    public int compareTo(Weight o) {
+        return (int) Math.ceil(Double.parseDouble(this.value) - Double.parseDouble(o.value));
+    }
```

Comment:
This could be unsafe, as there is no guarantee right now the weight is surely small enough to be cast to an integer; maybe you could try something like:

`return Double.compare(Double.parseDouble(this.value), Double.parseDouble(o.value));`

On another note, given that we have to parse the value of weight, we likely have to restrict the range of values we accept for weight to be within range of a double. This was something I overlooked when refactoring Weight class previously.